apps/order/schema.py

```diff
@@ -96,6 +98,74 @@ class Meta:
         filterset_class = OrderFilterSet
 
 
+class OrderStatisticType(graphene.ObjectType):
+    id = graphene.Int()
+    total_price = graphene.Int()
+    total_quantity = graphene.Int()
+
+
+class OrderStatType(graphene.ObjectType):
+    books_uploaded_count = graphene.Int(description='Total books upload count')
+    orders_completed_count = graphene.Int(description='Total orders completed count in last 3 months')
+    books_ordered_count = graphene.Int(description='Total books ordered count in last 3 months')
+    stat = CustomDjangoListField(OrderStatisticType)
+
+    class Meta:
+        fields = ()
+
+    @staticmethod
+    def resolve_books_uploaded_count(root, info, **kwargs):
+        '''
+        Returns total books uploaded count
+        '''
+        if info.context.user.user_type == User.UserType.ADMIN.value:
+            return Book.objects.count()
+        elif info.context.user.user_type == User.UserType.PUBLISHER.value:
+            return Book.objects.filter(publisher=info.context.user.publisher).count()
+        return Book.objects.filter(created_by=info.context.user).count()
+
+    @staticmethod
+    def resolve_orders_completed_count(root, info, **kwargs):
+        '''
+        Returns total orders completed in last 3 months
+        '''
+        stat_to = datetime.date.today()
+        stat_from = stat_to - datetime.timedelta(90)
+        return root.filter(
+            status=Order.OrderStatus.COMPLETED.value,
+            order_placed_at__gte=stat_from,
+            order_placed_at__lte=stat_to
+        ).count()
+
+    @staticmethod
+    def resolve_books_ordered_count(root, info, **kwargs):
+        '''
+        Returns total books ordered in last 3 months
+        '''
+        stat_to = datetime.date.today()
+        stat_from = stat_to - datetime.timedelta(90)
+        return root.filter(
+            status=Order.OrderStatus.COMPLETED.value,
+            order_placed_at__gte=stat_from,
+            order_placed_at__lte=stat_to
+        ).count()
+
+    @staticmethod
+    def resolve_stat(root, info, **kwargs):
+        '''
+        Returns order stat of in last 3 months
+        '''
+        stat_to = datetime.date.today()
+        stat_from = stat_to - datetime.timedelta(90)
```

Comment:
Maybe use a function for this. Make sure all functions get the same range.

```python3
def get_stat_daterange():
    stat_to = datetime.date.today()
    return stat_to - datetime.timedelta(90), stat_to
```

Reply:
@thenav56  I changed this